master/buildbot/test/unit/test_db_migrate_versions_024_add_buildslaves_table.py

```diff
@@ -42,9 +42,15 @@ def verify_thd(conn):
             self.assertEqual(res.fetchall(), [])
 
             # and buildslave name is unique, so we'll get an error here
-            dialect = conn.dialect.name
-            exc = (sa.exc.ProgrammingError if dialect == 'postgresql'
-                   else sa.exc.IntegrityError)
+            if conn.dialect.name == 'postgresql':
```

Comment:
I believe constraint violation is [IntegrityError](https://www.python.org/dev/peps/pep-0249/#integrityerror):

> Exception raised when the relational integrity of the database is affected, e.g. a foreign key check fails.

[ProgrammingError](https://www.python.org/dev/peps/pep-0249/#programmingerror):

> Exception raised for programming errors, e.g. table not found or already exists, syntax error in the SQL statement, wrong number of parameters specified, etc.